experimental/dds/tree2/src/feature-libraries/modular-schema/modularChangeFamily.ts

```diff
@@ -793,6 +819,16 @@ export class ModularChangeFamily
 			}
 		}
 
+		// If there's a node exists constraint and we deleted the node, increment the violation count
+		if (rebasedChange.nodeExistsConstraint !== undefined && deleted === true) {
+			// Only increment the violation count if the constraint wasn't already violated
+			// TODO: Decrement if constraint is fixed by rebasing (node is revived)
+			if (rebasedChange.nodeExistsConstraint.violated !== true) {
```

Comment:
```suggestion
			if ((rebasedChange.nodeExistsConstraint.violated ?? false)) {
```

I find this pattern to be slightly clearer as it makes it obvious that `violated` can be undefined and that its default value is false. Alternatively we could make an `isNodeExistenceConstraintViolated` helper.